rectools/models/rank.py

```diff
@@ -149,18 +151,23 @@ def _rank_on_gpu(
         object_norms: tp.Optional[np.ndarray],
         filter_query_items: tp.Optional[tp.Union[sparse.csr_matrix, sparse.csr_array]],
     ) -> tp.Tuple[np.ndarray, np.ndarray]:  # pragma: no cover
-        object_factors = implicit.gpu.Matrix(object_factors.astype(np.float32))
+
+        def _convert_arr_to_implicit_gpu_matrix(arr: np.ndarray) -> implicit.gpu.Matrix:
+            # We need to explicitly create copy to handle transposed arrays correctly
```

Comment:
```suggestion
            # We need to explicitly create copy to handle transposed arrays correctly
            # since Matrix is created from a direct copy of the underlying memory block, and `.T` is just a view
```

Reply:
fixed